src/sql/workbench/services/connection/browser/cmsConnectionController.ts

```diff
@@ -36,8 +36,8 @@ export class CmsConnectionController extends ConnectionController {
 		}, providerName);
 	}
 
-	public showUiComponent(container: HTMLElement): void {
+	public showUiComponent(container: HTMLElement, authTypeChanged: boolean = false): void {
 		this._databaseCache = new Map<string, string[]>();
```

Comment:
So instead of passing around this bool I think it'd be better to pass in the whole set of options and then the implementation can check those as necessary. This gets rid of the really confusingly named param here and also makes it easier in the future to do similar things where we want to control how the connection dialog looks.

Reply:
sending in all of the options doesn't make sense because we know we aren't going to use every field. But if that's something you want really badly, I can get that in later.

Reply:
I would say the way you currently have it with a very specific use-case and incorrectly-named parameter is worse. What downsides do you see to passing in the options?

Reply:
This is CMS specific and that's why has a very specific use case. If we want to use options, we should have at least a defined interface for it or one could send whatever in options because it's just a dictionary with string keys.

Reply:
Which is why I want to use something more generic. We shouldn't be adding CMS-specific code to a general object like the Connection Dialog. What happens when we add another provider type in the future that also wants to control how things appear? Then we have to add more params to these functions? That's very unsustainable. 

Having the options passed in means that callers just need to set the options and then create a new subclass like you did that reads in the appropriate options.

Reply:
This is a cmsConnectionController class, so having specific cases for a specific class makes sense to me. For a more generic handling connection dialog, it comes back to refactoring the dialog to make it more generalized. If we looked for options, what if 2 extension connection providers send 2 different values of auth changed per provider? This won't work in the current implementation.